File: C++/SearchAlgorithms/LinearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int size, int value){
    for(int i=0; i<size; i++){
        if(arr[i]==value){
            return i;
            break;
        }
    }
    return -1;
}

int main(){
    int size;
    cout<<"Enter the size of the array: ";
    cin>>size;
    int array[size];
    cout<<"Enter the elements: "<<endl;
    for(int i=0; i<size; i++){
        cin>>array[i];
    }
    cout<<"The elements are: "<<endl;
    for(int i=0; i<size; i++){
        cout<<array[i]<<" ";
    }
    int key;
    cout<<"\nEnter the value you want to search: ";
    cin>>key;

    int position = linearSearch(array, size, key);
    if(position==-1) cout<<"Not found!!!"<<endl;
    else cout<<"Found at position: "<<position+1<<endl;
    return 0;
}

File: Python/SearchAlgorithms/LinearSearch.py
def linearSearch(arr, size, value):
    print("Performing Linear Search: ")
    for i in range(size):
        if arr[i]==value:
            print(f"Found at position: {i+1}")
        


n = int(input("Enter the size of array: "))
print("Enter values in the array: ")

array = []

for i in range(n):
    j = int(input(f"value {i+1}: "))
    array.append(j)

print(f"The array is: {array}")
value = int(input("Enter the value you want to search in the array: "))
linearSearch(array, n, value)

